src/stream/src/executor/sync_kv_log_store.rs

```diff
@@ -530,38 +530,39 @@ impl<S: StateStore> SyncedKvLogStoreExecutor<S> {
         truncation_offset: Option<ReaderTruncationOffsetType>,
         buffer: &mut SyncedLogStoreBuffer,
     ) -> StreamExecutorResult<LogStorePostSealCurrentEpoch<'a, S::Local>> {
+        // TODO(kwannoel): As an optimization we can also change flushed chunks to be flushed items
+        // to reduce memory consumption of logstore.
+
         let epoch = barrier.epoch.prev;
         let mut writer = write_state.start_writer(false);
         writer.write_barrier(epoch, barrier.is_checkpoint())?;
 
-        // FIXME(kwannoel): Flush all unflushed chunks
-        // As an optimization we can also change it into flushed items instead.
-        // This will reduce memory consumption of logstore.
-
-        // TODO: may stop the for loop when seeing any of flushed item to avoid always iterating the whole buffer
-        for (epoch, item) in &mut buffer.buffer {
-            match item {
-                LogStoreBufferItem::StreamChunk {
-                    chunk,
-                    start_seq_id,
-                    end_seq_id,
-                    flushed,
-                    ..
-                } => {
-                    if !*flushed {
-                        writer.write_chunk(chunk, *epoch, *start_seq_id, *end_seq_id)?;
-                        *flushed = true;
+        if barrier.is_checkpoint() {
+            for (epoch, item) in &mut buffer.buffer {
```

Comment:
It seems that when we add items to `buffer`, we use `buffer.push_back(...)`, which means a newer item is at the back. Therefore, should we iterate from back to front instead?